Fix issue with invalid remote filenames

HelgeSverre opened the pull request:
I came across an issue where if the remote filename contained invalid characters (asterisk in my case) the local filename of the local file that you would download the remote file into would be invalid and you would get an error telling you the file does not exist.

aelvan commented:
Thanks for this! But I think maybe the sanitazing has to be done when the filename is defined, not in getFilePath. But I'll have to test this a little bit when I get the chance.

aelvan commented:
Great, I think maybe it would be sufficient to apply a fix inside `getPathsForUrl`, where I create the basename. I already change spaces to `-` there, maybe that's something that's already handled by the sanitize helper, so maybe you could just swap that out there.

Would be sweet if you would be able to create a new PR with that. :)

HelgeSverre commented:
Will take a look, thanks.

On Wed, 10 Oct 2018, 17:34 André Elvan, <notifications@github.com> wrote:

> Great, I think maybe it would be sufficient to apply a fix inside
> getPathsForUrl, where I create the basename. I already change spaces to -
> there, maybe that's something that's already handled by the sanitize
> helper, so maybe you could just swap that out there.
>
> Would be sweet if you would be able to create a new PR with that. :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/aelvan/Imager-Craft/pull/202#issuecomment-428620325>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABCgdAT0tvuIxFZfMJqWWpa9pSGorKo9ks5ujhOPgaJpZM4XUut8>
> .
>

HelgeSverre commented:
I reverted the previous change and moved the sanitizeFilename to the getPathsForUrl method. that method also seems to strip out whitespace and replace it with a dash in line 175 in the FileHelper.

HelgeSverre commented:
Does this solution look ok, or are there any changes you would make to it?

aelvan commented:
This has been merged into 2.1.1. Thanks for contributing!
